### ISSUE TITLE
Whether kubeadm(v1.15.3) init supports ipv6

### ISSUE THREAD

**Issue by Forest-L** [kubernetes/kubeadm#2003]
Versions:
kubeadm verison: v1.15.3
kubectl version: v1.15.3
OS: Centos 7.7
Kernel: 3.10.0-1062.el7.x86_64

what happened?
```
kubeadm init --config=kubeadm.yaml
[init] Using Kubernetes version: v1.15.3
[preflight] Running pre-flight checks
        [WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup driver. The recommended driver is "systemd". Please follow the guide at https://kubernetes.io/docs/setup/cri/
        [WARNING SystemVerification]: this Docker version is not on the list of validated versions: 19.03.3. Latest validated version: 18.09
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR HTTPProxy]: parse https://2402:e7c0:0:a00:ffff:****:****:fffb: invalid port ":fffb" after host
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
```
K8sv1.17.0 is ok, However K8sv1.15.3 is failed .

**Comment by rosti** [comment]
Yep, it's a problem with the HTTPProxy check itself. You can skip this check in particular to be able to continue forward.

The problem itself was fixed in kubernetes/kubernetes#82267 but that one was never cherry-picked into 1.15.

xref #1769
/priority important-longterm
/kind bug
/assign

**Comment by neolit123** [comment]
> K8sv1.17.0 is ok, However K8sv1.15.3 is failed .

if you write `https://[2402:e7c0:0:a00:ffff:****:****:fffb]` (surrounded by `[ ]`) does it work?
square brackets are standard practices for IPv6.

**Comment by Forest-L** [comment]
add --ignore-preflight-errors=HTTPProxy is ok, Thanks!

**Comment by rosti** [comment]
> > K8sv1.17.0 is ok, However K8sv1.15.3 is failed .
> 
> if you write `https://[2402:e7c0:0:a00:ffff:****:****:fffb]` (surrounded by `[ ]`) does it work?
> square brackets are standard practice for IPv6.

The field in the config requires an address, not an URL.

**Comment by Forest-L** [comment]
OK，kubeadm.yaml include an address.

**Comment by neolit123** [comment]
we decided to not apply the fix for 1.15 because it does not fit our backport criteria:
https://github.com/kubernetes/kubernetes/pull/82267

1.15 will be out of support in a couple of months when 1.18 is released.

@Forest-L 

> --ignore-preflight-errors=HTTPProxy

yes, this is a valid workaround.

/close
